### PR TITLE
Clarify how to disable key path spending

### PR DESCRIPTION
**Pull request by jonasnick**

### CODE REVIEW

**Comment by sipa** [comment]
ack

**Comment by real-or-random** [comment]
Shouldn't this be a random key for privacy?

**Comment by jonasnick** [comment]
@real-or-random yes, it is
> it is recommended to pick a fresh integer ''r'' in the range ''0...n-1'' uniformly at random and use ''H + rG'' as internal key

**Comment by real-or-random** [comment]
Nevermind, I'm too stupid to read.

**Comment by jonasnick** [comment]
Updated the link to how to derive H to a newer commit which made the code snippet actually copy-and-pastable.

**Comment by real-or-random** [comment]
ACK

File: bip-taproot.mediawiki [hunk 1]
```diff
@@ -170,7 +170,7 @@ Satisfying any of these conditions is sufficient to spend the output.
...
...e of the spending conditions consist of just a single key (after aggregation), the most likely one should be made the internal key. If no such condition exists, it may be worthwhile adding one that consists of an aggregation of all keys participating in all scripts combined; effectively adding an "everyone agrees" branch. If that is inacceptable, pick as internal key a point with unknown discrete logarithm. One example of such a point is ''H = point(0x0250929b74c1a04954b78b4b6035e97a5e078a5a0f28ec96d547bfee9ace803ac0)'' which is [https://github.com/ElementsProject/secp256k1-zkp/blob/11af7015de624b010424273be3d91f117f172c82/src/modules/rangeproof/main_impl.h#L16 constructed] by taking the hash of the uncompressed DER serialization of secp256k1 generator ''G'' as X coordinate. In order to avoid leaking the information that key path spending is not possible it is recommended to pick a fresh integer ''r'' in the range ''0...n-1'' uniformly at random and use ''H + rG'' as internal key. It is possible to prove that this internal key is does not have a known discrete logarithm with respect to ''G'' by revealing ''r'' to a verifier who can then reconstruct how the internal key was created.
```
Comment (sipa) [inline bip-taproot.mediawiki]: Nit: the elliptic curve point serialization is not DER; it's just the standard encoding for points. DER is only used for ECDSA signatures (and I hope nobody ever uses it for anything else...).

Comment (jonasnick) [reply bip-taproot.mediawiki]: Sorry I just mindlessly copied the description from secp-zkp. Fixed.

**Pull request merged**